account_move_accrual_entry/models/account_move.py

```diff
@@ -0,0 +1,278 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    accrual_date = fields.Date(string="Accrual Date")
+    accrual_account_move_ids = fields.Many2many(
+        comodel_name="account.move",
+        relation="accrual_account_move_rel",
+        column1="move_id",
+        column2="accrual_account_move_ids",
+        copy=False,
+        readonly=True,
+    )
+
+    def _get_accrual_account_id(self):
+        if self.accrual_date:
+            if not self.env.company.accrual_account_id:
+                raise UserError(
+                    _("Please set the accrual account in the invoincing settings.")
+                )
+        return self.env.company.accrual_account_id
+
+    def _get_accrual_asset_account_type_id(self):
+        if not self.env.company.accrual_asset_account_type_id:
+            raise UserError(
+                _("Please set the account type for assets in the invoincing settings.")
+            )
+        return self.env.company.accrual_asset_account_type_id
+
+    def _not_asset(self, x):
+        asset = self.env.company.accrual_asset_account_type_id.name
+        return x.accrual_account_id.user_type_id.name != asset
```

Comment:
We have to use the recordset itself, neither the name  nor the id, just the recordset `x.accrual_account_id.user_type_id != self.env.company.accrual_asset_account_type_id `